src/main/resources/config/liquibase/changelog/20210920225828_changelog.xml

```diff
@@ -0,0 +1,103 @@
+                LEFT JOIN text_block b ON s.id = b.submission_id AND f.reference = b.id
+            WHERE
+                e.discriminator = 'T'
+                AND f.reference IS NOT NULL
+                AND f.reference NOT IN (
+                    SELECT
+                    id FROM text_block
+                );
+
+            <!--This query finds all feedbacks which do not belong to any participation however they
+            are used for tutorial purposes-->
+            INSERT INTO feedbacks_to_migrate (text, start_index, end_index, submission_id, TYPE, feedback_id, tb_id)
+            SELECT
+                f.reference,
+                POSITION(f.reference IN s.text),
+                POSITION(f.reference IN s.text) + LENGTH(f.reference),
+                r.submission_id,
+                'MANUAL',
+                f.id,
+                SHA1(CONCAT(r.submission_id, ';', POSITION(f.reference IN s.text), '-', POSITION(f.reference IN s.text) + LENGTH(f.reference), ';', f.reference))
+            FROM
+                feedback f
+                INNER JOIN result r ON f.result_id = r.id
+                INNER JOIN submission s ON r.submission_id = s.id
+            WHERE
+                r.participation_id IS NULL
+                AND s.participation_id IS NULL
+                AND length(f.reference) > 255;
+
+            <!--This query creates all missing text blocks-->
+            INSERT INTO text_block (id, text, start_index, end_index, submission_id, TYPE)
+            SELECT
+                ftm.tb_id,
+                ftm.text,
+                ftm.start_index,
+                ftm.end_index,
+                ftm.submission_id,
+                ftm.type
+            FROM
+                feedbacks_to_migrate ftm;
+
+            <!--This query updates all feedback references-->
+            UPDATE
+                feedback f
+                INNER JOIN feedbacks_to_migrate ftm ON f.id = ftm.feedback_id
+            SET
+                f.reference = ftm.tb_id
+            WHERE
+                f.id = ftm.feedback_id;
+
+            <!--These last two queries find and delete all left over feedbacks which haven't been removed before-->
+            CREATE TEMPORARY TABLE leftover_feedbacks
+            SELECT
+                f.id AS feedback_id
+            FROM
+                feedback f
+                INNER JOIN result r ON f.result_id = r.id
+            WHERE
+                r.participation_id IS NULL
+                AND r.submission_id IS NULL;
+
+            DELETE FROM feedback
+            WHERE id IN (
+                SELECT
+                    feedback_id FROM leftover_feedbacks
+                )
+                OR result_id IS NULL;
+        </sql>
+    </changeSet>
+    <changeSet author="argertboja" id="20210920225828-2">
+        <!--Shorten the reference column length to 255 chars-->
+        <modifyDataType tableName="feedback" columnName="reference" newDataType="varchar(255)"/>
```

Comment:
did you check if there is feedback in the table already that exceed 255 chars?

Reply:
And is that restriction somehow reflected by the UI?

Reply:
Yes, I checked it and there are no more feedbacks exceeding 255 chars. 

What do you mean reflected by the UI?

Reply:
Perfect that every feedback is < 255 chars!
I mean, that a tutor/instructor is aware of not writing more than 255 chars, e.g. through any hint in the UI.

Reply:
But I guess as there are not entries > 255 one would not do that normally anyways... It just came to my mind, that it would be good to somehow show this limitation in the UI as well